### PR TITLE
Feature/particles

### PR DESCRIPTION
**Pull request by alvaro-soppelsa**
Particle System: second iteration
This adds the following new features:

- Particles start rotation and rotation over time
- Particle forces in each direction
- Animation blend between tiles
- Animation fit to lifetime
- New shapes of emission: circle and rectangle

In addition, ComponentBillboard has been refactored to match UI, code and features with particle system.
And some bugs were fixed too.

### CODE REVIEW

**Review by Izenz — APPROVED** [review]
Everything I tested seems to be working perfectly. Great job!

File: Source/src/components/ComponentParticleSystem.cpp [hunk 1]
```diff
@@ -565,6 +574,13 @@ void Hachiko::ComponentParticleSystem::UpdateEmitterTimes()
     time += EngineTimer::delta_time;
     emitter_elapsed_time += EngineTimer::delta_time;
 
+    if (emitter_elapsed_time < start_delay.GetValue())
+    {
+        able_to_emit = false;
+        return;
+    }
+
+    // constexpr float delta = 1-
```
Comment (baransrc) [inline Source/src/components/ComponentParticleSystem.cpp:583]:
```suggestion
```

File: Source/src/components/ComponentBillboard.cpp [hunk 2]
```diff
@@ -40,39 +39,41 @@ void Hachiko::ComponentBillboard::Draw(ComponentCamera* camera, Program* program
...
         gradient->getColorAt(color_frame, color.ptr());
     }
 
     program->BindUniformFloat4("input_color", color.ptr());
     program->BindUniformInts("has_texture", 1, &has_texture);
 
-    int flip_x = has_flip_x ? 1 : 0;
-    int flip_y = has_flip_y ? 1 : 0;
+    int flip_x = flip_texture.x ? 1 : 0;
+    int flip_y = flip_texture.y ? 1 : 0;
```
[the remark this answers is not in the document]
Comment (alvaro-soppelsa) [reply Source/src/components/ComponentBillboard.cpp]: Changed. Much better thanks!

File: Source/src/components/ComponentBillboard.h [hunk 3]
```diff
@@ -7,28 +7,18 @@ struct ImGradientMark;
...
         ComponentBillboard(GameObject* container);
         ~ComponentBillboard() override;
         
         void Start() override;
         void Update() override;
+
         void Play();
+        void Pause();
+        void Restart();
         void Stop() override;
```
[the remark this answers is not in the document]
Comment (alvaro-soppelsa) [reply Source/src/components/ComponentBillboard.h]: Ok! I added Stop too for something we just talk with Eric

File: Source/src/components/ComponentParticleSystem.cpp (hunk 1 above)
Comment (alvaro-soppelsa) [reply Source/src/components/ComponentParticleSystem.cpp]: Ups. Thanks

File: Source/src/utils/RandomUtil.cpp [hunk 4]
```diff
@@ -29,3 +29,8 @@ float Hachiko::RandomUtil::RandomBetween(const float2& values)
 {
     return RandomBetween(values.x, values.y);
 }
+
+int Hachiko::RandomUtil::RandomSignedInt()
```
[the remark this answers is not in the document]
Comment (alvaro-soppelsa) [reply Source/src/utils/RandomUtil.cpp]: Changed!

[the remark this answers is not in the document]
Comment (armandojaga) [reply Source/src/utils/RandomUtil.cpp]: I saw this too late, but I will change this to RandomInt, RandomSign sounds like you would get a - or + symbol

**Pull request merged**